packages/app/src/templates/services/template.service.test.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { TemplateService } from "App/templates/services/template.service"
+import { TemplateRepository } from "App/templates/repositories"
+import DeviceService from "Backend/device-service"
+import {
+  ErrorRequestResponse,
+  RequestResponseStatus,
+  SuccessRequestResponse,
+} from "App/core/types/request-response.interface"
+import { TemplatePresenter } from "App/templates/presenters"
+import { NewTemplate } from "App/templates/dto"
+
+jest.mock("App/templates/presenters")
+;(TemplatePresenter as unknown as jest.Mock).mockImplementation(() => {
+  return {
+    mapToPureTemplateBody: jest.fn(),
+    mapToTemplate: jest.fn(),
+  }
+})
+
+const templateRepository = {
+  create: jest.fn(),
+  update: jest.fn(),
+  delete: jest.fn(),
+} as unknown as TemplateRepository
+
+const deviceService = {
+  request: jest.fn(),
+} as unknown as DeviceService
+
+const subject = new TemplateService(deviceService, templateRepository)
+
+const successResponse: SuccessRequestResponse<any> = {
+  status: RequestResponseStatus.Ok,
+  data: {},
+}
+
+const errorResponse: ErrorRequestResponse = {
+  status: RequestResponseStatus.Error,
+}
+
+const template: NewTemplate = {
+  text: "Hello world",
+}
+
+beforeEach(() => {
+  jest.resetAllMocks()
+})
+
+describe("`TemplateService`", () => {
+  describe("`createTemplate` method", () => {
+    test("map data and returns success when `deviceService.request` returns success", async () => {
+      deviceService.request = jest.fn().mockReturnValue(successResponse)
+      const response = await subject.createTemplate(template)
+      expect(deviceService.request).toHaveBeenCalled()
+      expect(TemplatePresenter.mapToPureTemplateBody).toHaveBeenCalled()
+      expect(TemplatePresenter.mapToTemplate).toHaveBeenCalled()
```

Comment:
Imho it is just an implementation detail that should not be tested in that case. Maybe:
1. we should check what was returned by the function?
2. if really needed, maybe it would be better just to check what body was passed to the `deviceService.request` method (instead of checking if `TemplatePresenter.mapToPureTemplateBody` was called)?

Reply:
https://github.com/mudita/mudita-center/pull/746/commits/774bf9c9ba9adb5a6c07b1a8245b1baecbd9d376